File: journal-service/src/main/java/kioke/journal/constant/Role.java
package kioke.journal.constant;

import java.util.Set;

public enum Role {
  AUTHOR(Set.of(Permission.READ, Permission.WRITE, Permission.DELETE, Permission.SHARE)),
  EDITOR(Set.of(Permission.READ, Permission.WRITE)),
  READER(Set.of(Permission.READ));

  private final Set<Permission> permissions;

  private Role(Set<Permission> permissions) {
    this.permissions = permissions;
  }

  public boolean hasPermission(Permission permission) {
    return permissions.contains(permission);
  }
}

File: web-client/src/constants/role.ts
export enum Role {
  AUTHOR = 'AUTHOR',
  EDITOR = 'EDITOR',
  READER = 'READER',
}

export const Roles: {
  [key in Role]: {
    title: string;
  };
} = {
  [Role.AUTHOR]: {
    title: 'Author',
  },
  [Role.EDITOR]: {
    title: 'Editor',
  },
  [Role.READER]: {
    title: 'Reader',
  },
};
